perfmetrics/scripts/vm_metrics/vm_metrics.py

```diff
@@ -100,133 +98,179 @@ def _validate_start_end_times(self, start_time_sec, end_time_sec):
-      factor (float) : The factor by which the value of API response should be
-                       divided to get inot desired units.
+      instance (str): VM instance name
+      period (float): Period over which the values are aligned
+      metric_type (str): The type of metric
+      factor (float) : Converting the API response values into required units.
+      aligner(str): Operation to be applied at points of each period
+      reducer(str): Operation to aggregate data points accross multiple metrics
+      group_fields(list[str]): The fields we want to aggregate using reducer
     Returns:
       list[MetricPoint]
     """
-    peak_metrics_response, mean_metrics_response = self._get_api_response(
-        metric_type, start_time_sec, end_time_sec, instance, period)
-    metrics_data = _create_metric_points_from_response(
-        peak_metrics_response, mean_metrics_response, factor)
+    metrics_response = self._get_api_response(metric_type, start_time_sec,
+                                              end_time_sec, instance, period,
+                                              aligner, reducer, group_fields)
+    metrics_data = _create_metric_points_from_response(metrics_response, factor)
+    
+    # In case OPS_ERROR_COUNT data is empty, we return a list of zeroes:
+    if(metric_type == OPS_ERROR_COUNT_METRIC and metrics_data == []):
+      return [MetricPoint(0, 0, 0) for i in range(int((end_time_sec-start_time_sec)/period)+1)]
+    
+    # Metrics data for metrics other that OPS_ERROR_COUNT_DATA should not be empty:
+    if(metrics_data == []):
+      raise NoValuesError('No values were retrieved from the call')
 
     return metrics_data
 
   def fetch_metrics_and_write_to_google_sheet(self, start_time_sec,
                                               end_time_sec, instance,
-                                              period) -> None:
-    """Fetches the metrics data for cpu utilization and received bytes count and writes it to a google sheet.
+                                              period, test_type) -> None:
+    """Fetches the metrics data for all types and writes it to a google sheet.
 
     Args:
       start_time_sec (int): Epoch seconds
       end_time_sec (int): Epoch seconds
       instance (str): VM instance
-      period (float): Period over which the mean and peak values are taken
+      period (float): Period over which the values are taken
+      test_type(str): The type of load test for which metrics are taken
     Returns: None
     """
     self._validate_start_end_times(start_time_sec, end_time_sec)
-    cpu_uti_data = self._get_metrics(start_time_sec, end_time_sec, instance,
-                                     period, CPU_UTI_METRIC, 1 / 100)
-    rec_bytes_data = self._get_metrics(start_time_sec, end_time_sec, instance,
-                                       period, RECEIVED_BYTES_COUNT_METRIC,
-                                       60000)
+    global TEST_TYPE
```

Comment:
you shouldn't declare global variables like this. 
test_type is anyways a param here, pass the same to required function and then call a method like this whereever you need this conversion. _get_ops_metric_method_name(test_type)

Reply:
done